### ISSUE TITLE
Please establish lossless and secure compression as the primary idea

### ISSUE THREAD

**Issue by yisibl** [svg/svgo#1360]
As an svg compressor, I think it is most important not to change the display effect of the svg itself, which means that the compressed result cannot change the shape, position, size, color, etc. (svgcleaner has a [detailed description](https://github.com/RazrFalcon/svgcleaner/blob/master/docs/testing_notes.rst#rendering-errors)). As svgo is adopted by more and more non-browser terminals ([Sketch](https://github.com/sketch-hq/svgo-compressor), Adobe Illustrator, other back-end languages), this will become more and more difficult, because each side implements svg differently. So we urgently need to introduce Correctness test like [svgcleaner](https://github.com/RazrFalcon/svgcleaner#correctness) to help svgo become the best svg compressor.

The following work is a good start:

- [x]  [The `noSpaceAfterFlags` option is disabled by default](https://github.com/svg/svgo/pull/1353) @yisibl 
- [x] [Test regressions with W3C SVG 1.1 Test Suite](https://github.com/svg/svgo/pull/1357) @TrySound 
- [ ] Add sample documents and compatibility instructions for each option of svgo, Similar to:
![image](https://user-images.githubusercontent.com/2784308/108680243-942c3980-7528-11eb-919d-b04950d1ce2d.png)

cc @deepsweet

**Comment by sk-** [comment]
May I add that we should change the default options to only include those plugins known/supposed to be lossless. For example `convertPathData` probably should not be a default plugin unless the default options are safe. I've also seen that gradients are usually changed.

**Comment by TrySound** [comment]
Or we should make it safe.

**Comment by mischnic** [comment]
Another case here is `cleanupIds`, which is generally good, unless you use something like `<use xlink:href="square.svg#square" x="20" fill="white" stroke="red"/>` where removing the IDs inside `square.svg` isn't safe.

**Comment by TrySound** [comment]
We have now presets concept. I'm ok to implement different presets. Though would be good if somebody explore all plugins.
I'm currently busy preparing svgo 3.

**Comment by SethFalco** [comment]
Closing this as we're now actively striving for this, but the issue isn't presented in a way that we can track progress and know when it's complete.

If you have concrete suggestions, you're encouraged to open another issue with what we can specifically do better. Meanwhile, I'm not at all dismissing the issue.

All future development is done with lossless compression in mind. We're investing a lot of time and effort in fixing bugs, improving testing, and having more complete documentation to clarify potential quirks. I'm also working on new sections to include in the documentation to clarify what we optimize, and in what cases we may deem something more important than reducing bytes.

In recent released, I believe we've already made good progress, and will continue to do so. 👍🏽
